<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <title>The source code</title>
  <link href="../resources/prettify/prettify.css" type="text/css" rel="stylesheet" />
  <script type="text/javascript" src="../resources/prettify/prettify.js"></script>
  <style type="text/css">
    .highlight { display: block; background-color: #ddd; }
  </style>
  <script type="text/javascript">
    function highlight() {
      document.getElementById(location.hash.replace(/#/, "")).className = "highlight";
    }
  </script>
</head>
<body onload="prettyPrint(); highlight();">
  <pre class="prettyprint lang-js"><span id='java-security-PrivilegedActionException'>/**@class java.security.PrivilegedActionException
</span>@extends java.lang.Exception

 Legacy security code; do not use.

 This exception is thrown by
 {@code doPrivileged(PrivilegedExceptionAction)} and
 {@code doPrivileged(PrivilegedExceptionAction,
 AccessControlContext context)} to indicate
 that the action being performed threw a checked exception.  The exception
 thrown by the action can be obtained by calling the
 {@code getException} method.  In effect, an
 {@code PrivilegedActionException} is a &quot;wrapper&quot;
 for an exception thrown by a privileged action.

 &lt;p&gt;As of release 1.4, this exception has been retrofitted to conform to
 the general purpose exception-chaining mechanism.  The &quot;exception thrown
 by the privileged computation&quot; that is provided at construction time and
 accessed via the {@link #getException}() method is now known as the
 &lt;i&gt;cause&lt;/i&gt;, and may be accessed via the {@link Throwable#getCause()}
 method, as well as the aforementioned &quot;legacy method.&quot;

 @see PrivilegedExceptionAction
 @see AccessController#doPrivileged(PrivilegedExceptionAction)
 @see AccessController#doPrivileged(PrivilegedExceptionAction,AccessControlContext)
*/
var PrivilegedActionException = {

<span id='java-security-PrivilegedActionException-method-getException'>/**Returns the exception thrown by the privileged computation that
</span> resulted in this {@code PrivilegedActionException}.

 &lt;p&gt;This method predates the general-purpose exception chaining facility.
 The {@link Throwable#getCause()} method is now the preferred means of
 obtaining this information.
@return {Object {java.lang.Exception}} the exception thrown by the privileged computation that
         resulted in this {@code PrivilegedActionException}.
@see PrivilegedExceptionAction
@see AccessController#doPrivileged(PrivilegedExceptionAction)
@see AccessController#doPrivileged(PrivilegedExceptionAction,
                                            AccessControlContext)
*/
getException : function(  ) {},

<span id='java-security-PrivilegedActionException-method-getCause'>/**Returns the cause of this exception (the exception thrown by
</span> the privileged computation that resulted in this
 {@code PrivilegedActionException}).
@return {Object {java.lang.Throwable}} the cause of this exception.
@since 1.4
*/
getCause : function(  ) {},

<span id='java-security-PrivilegedActionException-method-toString'>/**
</span>*/
toString : function(  ) {},


};</pre>
</body>
</html>
